plaidml2/bridge/keras/__init__.py

```diff
@@ -348,27 +437,34 @@ def categorical_crossentropy(target, output, from_logits=False):
     return _KerasNode('categorical_crossentropy', tensor=R)
 
 
+@_log_call
 def ceil(x):
-    _report_unimplemented('ceil')
+    return _KerasNode('ceil', tensor=edsl.ceil(x.tensor))
 
 
+@_log_call
 def clear_session():
-    _report_unimplemented('clear_session')
+    global _in_train_phase, _device
+    _in_train_phase = None
+    _device = None
```

Comment:
We don't really have any state to clear at the moment, but the MLIR-based EDSL might in the future. But setting `_device_ = None` will definitely break things.

Reply:
Instead re-getting the device ... not sure that matters. There was some issue that inspired us to put code here with the previous Keras backend, I'll see if I can dig it up and see whether its concerns apply...

Reply:
Ah, the issue I was thinking of wasn't super helpful: https://github.com/plaidml/plaidml/issues/132